mautic.dc.yaml

```diff
@@ -0,0 +1,149 @@
+kind: Template
+apiVersion: v1
+objects:
+  - apiVersion: v1
+    data:
+      Caddyfile: |
+          # Where caddy should listen
+          :2015
+
+          # Turn on the Web/file server
+          file_server
+
+          # The site root
+          root * /opt/app-root/src
+
+          # Because we should
+          encode zstd gzip
+
+          # This rewrite rule may be required to get React apps
+          # to deal with trailing slashes properly.
+          @filter {
+          	not path_regexp ^\/0.0.0.0
+          	file {
+          		try_files {path} {path}/ /?url={uri}
+          	}
+          }
+          # rewrite @try_files {http.matchers.file.relative}
+          try_files {path} {path}/ index.html
+
+          # On OCP we should log to stdout so Prometheus can
+          # slurp up the logs for human consumption.
+          log {
+            #errors stdout
+            output stdout 
+            format single_field common_log
+            level info
+          }
+        
+
+    kind: ConfigMap
+    metadata:
+      creationTimestamp: null
+      name: caddy-${NAME}${SUFFIX}
+      labels:
+        app: ${NAME}${SUFFIX}
+        deploymentconfig: ${NAME}${SUFFIX}
+        app-name: ${NAME}${SUFFIX}
+  - apiVersion: apps.openshift.io/v1
+    kind: DeploymentConfig
+    spec:
+      replicas: 1
+      selector:
+        deploymentconfig: ${NAME}
+      strategy:
+        resources: {}
+      template:
+        metadata:
+          annotations:
+            openshift.io/generated-by: OpenShiftNewApp
+          creationTimestamp: null
+          labels:
+            deploymentconfig: ${NAME}
+        spec:
+          containers:
+            - image: '${NAME}:latest'
+              name: ${NAME}
+              ports:
+                - containerPort: 2015
+                  protocol: TCP
+              resources: {}
+              volumeMounts:
+              - name: caddy-${NAME}${SUFFIX}
+                mountPath: /opt/app-root/etc/Caddyfile
+                readOnly: true
+                subPath: Caddyfile
+          volumes:
+          - name: caddy-${NAME}${SUFFIX}
+            configMap:
+              defaultMode: 420
+              name: caddy-${NAME}${SUFFIX}
+      test: false
+      triggers:
+        - type: ConfigChange
+        - imageChangeParams:
+            automatic: true
+            containerNames:
+              - ${NAME}
+            from:
+              kind: ImageStreamTag
+              name: '${NAME}:latest'
+          type: ImageChange
+    metadata:
+      annotations:
+        openshift.io/generated-by: OpenShiftNewApp
+      labels:
+        app: ${NAME}
+        app.kubernetes.io/component: ${NAME}
+        app.kubernetes.io/instance: ${NAME}
+      name: ${NAME}
+  - apiVersion: v1
+    kind: Service
+    spec:
+      ports:
+        - name: 2015-tcp
+          port: 2015
+          protocol: TCP
+          targetPort: 2015
+      selector:
+        deploymentconfig: ${NAME}
```

Comment:
this is the selector i was talking aobut it. IT must match dc.spec.template.spec.labels or something liek that